WooCommerce/Classes/Tools/VersionHelpers.swift

```diff
@@ -0,0 +1,135 @@
+            string.replacingOccurrences(of: "([^0-9.]+)", with: ".$1.", options: .regularExpression)
+        }
+
+        /// Score and compare two string components
+        ///
+        func compareStringComponents(_ lhs: String, _ rhs: String) -> ComparisonResult {
+            /// Score each component
+            let lhsScore = VersionComponentScore(from: lhs)
+            let rhsScore = VersionComponentScore(from: rhs)
+
+            if lhsScore < rhsScore {
+                return .orderedAscending
+            }
+
+            if lhsScore > rhsScore {
+                return .orderedDescending
+            }
+
+            if lhsScore == .number && rhsScore == .number {
+                let lhsAsNumber = NSNumber(value: Int(lhs) ?? 0)
+                let rhsAsNumber = NSNumber(value: Int(rhs) ?? 0)
+
+                let comparisonResult = lhsAsNumber.compare(rhsAsNumber)
+                if comparisonResult != .orderedSame {
+                    return comparisonResult
+                }
+            }
+
+            return .orderedSame
+        }
+
+        /// Process the given string into version components
+        ///
+        func versionComponents(of string: String) -> [String] {
+            var stringToComponentize = replaceUnderscoreDashAndPlusWithDot(string)
+            stringToComponentize = insertDotsBeforeAndAfterAnyNonNumber(stringToComponentize)
+            return stringToComponentize.components(separatedBy: ".")
+        }
+
+        let leftComponents = versionComponents(of: lhs)
+        let rightComponents = versionComponents(of: rhs)
+
+        let maxComponents = max(leftComponents.count, rightComponents.count)
+
+        for index in 0..<maxComponents {
+            /// Treat missing components (e.g. 1.2 being compared to 1.1.3 as "0", i.e. 1.2.0
+            let leftComponent = index < leftComponents.count ? leftComponents[index] : "0"
+            let rightComponent = index < rightComponents.count ? rightComponents[index] : "0"
+
+            let comparisonResult = compareStringComponents(leftComponent, rightComponent)
+            if comparisonResult != .orderedSame {
+                return comparisonResult
+            }
+        }
+
+        return .orderedSame
+    }
+}
+
+/// Defines the score (rank) of a component string within a version string.
+/// e.g. the "3" in 3.0.0beta3 should be treated as `.number`
+/// and the "beta" should be scored (ranked) lower as `.beta`.
+///
+/// The scores of components of version strings are used when comparing complete version strings
+/// to decide if one version is older, equal or newer than another.
+///
+/// Ranked per https://www.php.net/manual/en/function.version-compare.php
+///
+fileprivate enum VersionComponentScore: Comparable {
+    case other
+    case dev
+    case alpha
+    case beta
+    case RC
+    case number
+    case patch
+}
+
+fileprivate extension VersionComponentScore {
```

Comment:
nit: same as above, I think this could be private:

```suggestion
private extension VersionComponentScore {
```

Reply:
Fixed as part of https://github.com/woocommerce/woocommerce-ios/pull/5382/commits/2eb5cb0402c9bef48cd7b74fe3d81bbe031f3c44